app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,187 @@
+      begin
+        Dir.mkdir(data_dir)
+      rescue Errno::EEXIST
+        # Ignore if the directory was created by another thread
+        _log.debug("No need to create directory #{data_dir} since it exists.")
+      end
+    end
+    ost.skipConfig = true
+    ost.config     = OpenStruct.new(
+      :dataDir            => data_dir,
+      :forceFleeceDefault => false
+    )
+
+    begin
+      require 'blackbox/VmBlackBox'
+      _log.debug("instantiating BlackBox")
+      bb = Manageiq::BlackBox.new(vm.guid, ost)
+      _log.debug("instantiated BlackBox")
+      categories = AmazonSsaSupport::SsaQueueExtractor::CATEGORIES
+      #
+      # The amazon_ssa_support gem defines the categories as an array,
+      # But the scanning_mixin expects it to be a comma-separated string.
+      #
+      ost.category = categories.join(",")
+
+      categories.each do |c|
+        ost.taskid = job.guid
+        update_job_message(ost, "Syncing #{c}")
+        ost.taskid = ost.jobid
+        st = Time.current
+        xml = MIQRexml.load(ost.reply[:categories][c.to_sym])
+        if xml.nil?
+          _log.debug("No XML loaded for [#{c}].")
+          next
+        elsif xml.root.nil?
+          _log.debug("No XML root loaded for [#{c}]: XML is #{xml}.")
+          next
+        end
+        _log.debug("Writing scanned data to XML for [#{c}] to blackbox.")
+        bb.saveXmlData(xml, c)
+        _log.debug("Writing XML complete.")
+
+        category_node = xml_summary.class.load(xml.root.shallow_copy.to_xml.to_s).root
+        category_node.add_attributes("start_time" => st.utc.iso8601, "end_time" => Time.now.utc.iso8601)
+        xml_node << category_node
+        categories_processed += 1
+      end
+    rescue NoMethodError => scan_err
+      last_err = scan_err
+      _log.error("perform_metadata_sync Error - [#{scan_err}]")
+      _log.error("perform_metadata_sync Error - [#{scan_err.backtrace.join("\n")}]")
+    ensure
+      bb&.close
+      ost.taskid = job.guid
+      update_job_message(ost, "Scanning completed.")
+      ost.taskid = ost.jobid
+      if last_err
+        status = "Error"
+        status_code  = 8
+        status_code  = 16 if categories_processed.zero?
+        scan_message = last_err.to_s
+        _log.error("ScanMetadata error status:[#{status_code}]:  message:[#{last_err}]")
+        _log.debug { last_err.backtrace.join("\n") }
+      end
+
+      xml_node_scan.add_attributes(
+        "end_time"    => Time.now.utc.iso8601,
+        "status"      => status,
+        "status_code" => status_code.to_s,
+        "message"     => scan_message
+      )
+      vm.save_metadata_op(MIQEncode.encode(xml_summary.to_xml.to_s), "b64,zlib,xml", job.guid)
+      _log.info("Completed: Sending scan summary to server.  TaskId:[#{ost.taskid}]  target:[#{vm.name}]")
+    end
+  end
+
+  def response_thread_sleep_seconds
+    @response_thread_sleep_seconds ||= Settings.ems.ems_amazon.agent_coordinator.response_thread_sleep_seconds
```

Comment:
There's no need to memoize this value.  In fact, that's bad because if the user changes the Settings, you probably want to live pick up the value.  Getting a value out of Settings has almost no overhead, also, so the caching is unnecessary.

Reply:
Fixed in https://github.com/ManageIQ/manageiq-providers-amazon/pull/360